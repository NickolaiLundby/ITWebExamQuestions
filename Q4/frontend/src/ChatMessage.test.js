import React from 'react';
import ChatMessage from './ChatMessage';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

describe('<ChatMessage/>', () => {
    test('Renders', () => {
        const mockNameMsg = {
            name: 'Bob', 
            message: 'Hello'
        };
        const wrapper = shallow(<ChatMessage name={mockNameMsg.name} message={mockNameMsg.message} />);
        expect(wrapper.find('p').length).toBe(1);
        expect(wrapper.find('p').text()).toBe(mockNameMsg.name + " " + mockNameMsg.message);
        expect(wrapper.find('strong').length).toBe(1);
        expect(wrapper.find('strong').text()).toBe(mockNameMsg.name);
        expect(wrapper.find('em').length).toBe(1);
        expect(wrapper.find('em').text()).toBe(mockNameMsg.message);
    });
});

test('snapshot', () => {
    const mockNameMsg = {
        name: 'Alice', 
        message: 'Hello'
    };
    const component = renderer.create(<ChatMessage name={mockNameMsg.name} message={mockNameMsg.message} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})