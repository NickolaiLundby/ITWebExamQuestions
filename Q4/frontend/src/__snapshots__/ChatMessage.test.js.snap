// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<p>
  <strong>
    Alice
  </strong>
   
  <em>
    Hello
  </em>
</p>
`;
